Homework1/Task05/task05.py

```diff
@@ -7,23 +7,15 @@
     result = 16
 """
 from typing import List
-import itertools
-
-
-def contains_sublist(my_list, sublist):
-    n = len(sublist)
-    return any((sublist == my_list[i : i + n]) for i in range(len(my_list) - n + 1))
 
 
 def find_maximal_subarray_sum(nums: List[int], k: int) -> int:
     max_sum = 0
     for j in range(1, k + 1):
-        comb = [
-            i
-            for i in itertools.combinations(nums, j)
-            if contains_sublist(nums, list(i))
-        ]
-        for l in range(len(comb)):
-            if sum(comb[l]) > max_sum:
-                max_sum = sum(comb[l])
+        i = 0
+        while i + j <= len(nums):
+            comb = nums[i : i + j]
+            if sum(comb) > max_sum:
```

Comment:
`sum` is a costly operation (0(k)), so we can cache it in a variable to avoid calling twice. 

This not only optimizes performance but also improve readability.